jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -48,6 +50,42 @@ def _check_is_file(path):
     return isinstance(path, str) and Path(path).is_file()
 
 
+def _warn_if_not_found(app, file_lists):
+    """
+    Take a list of labels and associated file lists and send a snackbar
+    message if the length of any list is 0.
+    """
+    found = []
+    not_found = []
+    for key in file_lists:
+        if len(file_lists[key]) == 0:
+            not_found.append(key)
+        else:
+            found.append(key)
+
+    if len(found) == 0:
+        raise ValueError("No valid NIRISS files found in specified directory")
```

Comment:
Maybe remove "NIRISS" so that we could use this function for e.g. the NIRSpec parser as well? Or we could have the instrument be an optional parameter to the function and use it to populate this string.

Reply:
Good call! I implemented your suggestions here and in your comment.